pkg/ironic/ironic.go

```diff
@@ -0,0 +1,128 @@
+package ironic
+
+import (
+	"time"
+
+	metalkubev1alpha1 "github.com/metalkube/baremetal-operator/pkg/apis/metalkube/v1alpha1"
+	logf "sigs.k8s.io/controller-runtime/pkg/runtime/log"
+)
+
+var log = logf.Log.WithName("ironic")
+var deprovisionRequeueDelay = time.Second * 10
+
+// Provisioner implements the provisioning.Provisioner interface
+// and uses Ironic to manage the host.
+type Provisioner struct{}
```

Comment:
The common pattern we'll probably want to follow is to have this implementation of the interface non-exported, and have a constructor that returns the interface. That keeps the implementation fully non-public.

```
type provisioner struct{}

func New() provisioning.Provisioner {
    return provisioner{}
}
```

Then that gets called as:
```
myProvisioner := ironic.New()
```

Reply:
Fixed